core/trino-main/src/main/java/io/trino/execution/CreateFunctionTask.java

```diff
@@ -92,21 +92,20 @@ public ListenableFuture<Void> execute(CreateFunction statement, QueryStateMachin
 
         accessControl.checkCanCreateFunction(session.toSecurityContext(), name);
 
-        String formatted = formatSql(function);
-        verifyFormattedFunction(formatted, function);
-
-        languageFunctionManager.verifyForCreate(session, formatted, functionManager, accessControl);
+        languageFunctionManager.verifyForCreate(session, function, functionManager, accessControl);
 
         String signatureToken = languageFunctionManager.getSignatureToken(function.getParameters());
 
+        String sql = formattedFunction(function);
```

Comment:
Formatter may change in the future. Why format before storing?
Add a code comment.

Reply:
This is an AST node. The formatting is for serialization.

Reply:
I realize now that the `formatted` part of the name is not useful. I named it that originally because it calls `formatSql`. The fact that it's "formatting" in the "make the SQL look pretty" sense is not important. I renamed it to `functionToSql` as that reflects the purpose.